pkg/controller/compliancesuite/suitererunner.go

```diff
@@ -0,0 +1,112 @@
+package compliancesuite
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	compv1alpha1 "github.com/openshift/compliance-operator/pkg/apis/compliance/v1alpha1"
+	"github.com/robfig/cron"
+	batchv1 "k8s.io/api/batch/v1"
+	batchv1beta1 "k8s.io/api/batch/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/openshift/compliance-operator/pkg/utils"
+)
+
+const rerunnerServiceAccount = "rerunner"
+
+func (r *ReconcileComplianceSuite) reconcileScanRerunnerCronJob(suite *compv1alpha1.ComplianceSuite, logger logr.Logger) error {
+	rerunner := getRerunner(suite)
+	if suite.Spec.Schedule == "" {
+		return r.handleDelete(rerunner, logger)
+	}
+	// Verify that the Schedule is in a correct format
+	_, err := cron.ParseStandard(suite.Spec.Schedule)
+	if err != nil {
+		msg := "ComplianceSuite's schedule is wrongly formatted"
+		r.recorder.Event(
+			suite,
+			corev1.EventTypeWarning,
+			"ScheduleFormatError",
+			msg,
+		)
+		logger.Info("WARNING: " + msg)
+		// TODO(jaosorior): Should we surface this to the CRD?
```

Comment:
I would say yes. The admin has expressed intent to schedule the scans and the controller cannot converge to the state because the input was wrong. I would say this should be expressed in the status.

Reply:
The reason I didn't do this is that we currently don't have a field in the status where such warnings or errors would appear. Right now it's just an event. The other thing is that this is a non-fatal issue, a scan will still be executed, this just won't have scheduled scans afterwards... the Aggregated phase and results are probably not the places to put this.. so... any suggestions on what to add to reflect this?

Reply:
Oh, the suite status we have now is just an aggregate of scan statuses..but still, couldn't we (ab)use the aggregated phase and result and surface the error there?
I was thinking about this: if a schedule is specified, but cannot be parsed, then don't schedule new scans and kill the existing ones, then surface the error to the aggregated error and status fields.

Reply:
OK, that could be a way to go.

Reply:
Mind if I do this in a subsequent PR?

Reply:
sure, let me lgtm this patchset so that it merges when CI finishes